<a href="https://colab.research.google.com/github/macde58/Proyecto-Final/blob/main/Proyecto_Final_MAC.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

 PROYECTO FINAL  - PREDICCION DE PRECIOS DE CASAS 
#-----------------------------------------------------

## Miguel Angel Cardona de la Rosa
## Canet 08001084

**IMPORTACION DE LIBRERIAS**
### se cargan las librerias autolizadas a usar durante el proyecto


In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [4]:
import seaborn as sns

**CARGA DE DATA**

 Se carga a la estructura de este proyecto  el dataset dado para el proyecto

In [6]:
datafile = "proyecto_training_data.npy"
data= np.load(datafile)
print(data)

[[2.08500e+05 7.00000e+00 8.56000e+02 8.00000e+00 2.00300e+03 6.50000e+01]
 [1.81500e+05 6.00000e+00 1.26200e+03 6.00000e+00 1.97600e+03 8.00000e+01]
 [2.23500e+05 7.00000e+00 9.20000e+02 6.00000e+00 2.00100e+03 6.80000e+01]
 ...
 [2.66500e+05 7.00000e+00 1.18800e+03 9.00000e+00 1.94100e+03 6.60000e+01]
 [1.42125e+05 5.00000e+00 1.07800e+03 5.00000e+00 1.95000e+03 6.80000e+01]
 [1.47500e+05 5.00000e+00 1.25600e+03 6.00000e+00 1.96500e+03 7.50000e+01]]


**EXPLORACION DEL DATASET**

Se explora el dataset, validando su tipo de datos, shape y los valores nan que posea

In [7]:
print("INFORMACION DEL DATASET")
print("Shape:       ", data.shape)
print("Tipo:        ", type(data))
print("Valores Nan:  ", np.isnan(data).sum())

INFORMACION DEL DATASET
Shape:        (1460, 6)
Tipo:         <class 'numpy.ndarray'>
Valores Nan:   259


**LIMIPIEZA DEL DATASET**

En el paso anterior se determina que la estructura contiene 259 valoron "nan" , por lo que el siguiente paso se reasigan




In [8]:
data=data[~np.isnan(data).any(axis=1)]

In [9]:
print("INFORMACION DEL DATASET")
print("Shape:       ", data.shape)
print("Tipo:        ", type(data))
print("Valores Nan:  ", np.isnan(data).sum())

INFORMACION DEL DATASET
Shape:        (1201, 6)
Tipo:         <class 'numpy.ndarray'>
Valores Nan:   0


**EXPLORACION  DATAFRAME COMPLETO**

Ya habiendo reemplazado todos los datos nan, se utiliza esta estructura para presentar los datos solicitados que son:


*   Media
*   Max
*   Min
*   Deviacion Estandar 
*   Rango



In [10]:
## usando pandas para convertir lo s datos 
columnas = ["SalePrice","OverallQual","1stFlrSF","TotRmsAbvGrd","YearBuilt","LotFrontage"]
data_frame =pd.DataFrame(data,columns=columnas)
data_frame


SalePrice  OverallQual  1stFlrSF  TotRmsAbvGrd  YearBuilt  LotFrontage
0      208500.0          7.0     856.0           8.0     2003.0         65.0
1      181500.0          6.0    1262.0           6.0     1976.0         80.0
2      223500.0          7.0     920.0           6.0     2001.0         68.0
3      140000.0          7.0     961.0           7.0     1915.0         60.0
4      250000.0          8.0    1145.0           9.0     2000.0         84.0
...         ...          ...       ...           ...        ...          ...
1196   175000.0          6.0     953.0           7.0     1999.0         62.0
1197   210000.0          6.0    2073.0           7.0     1978.0         85.0
1198   266500.0          7.0    1188.0           9.0     1941.0         66.0
1199   142125.0          5.0    1078.0           5.0     1950.0         68.0
1200   147500.0          5.0    1256.0           6.0     1965.0         75.0

[1201 rows x 6 columns]

In [11]:
## Analisis estadistico
data_frame.describe()

SalePrice  OverallQual  ...    YearBuilt  LotFrontage
count    1201.000000  1201.000000  ...  1201.000000  1201.000000
mean   180770.480433     6.122398  ...  1970.580350    70.049958
std     83389.519866     1.419216  ...    31.750335    24.284752
min     34900.000000     1.000000  ...  1872.000000    21.000000
25%    127500.000000     5.000000  ...  1950.000000    59.000000
50%    159500.000000     6.000000  ...  1972.000000    69.000000
75%    213500.000000     7.000000  ...  2003.000000    80.000000
max    755000.000000    10.000000  ...  2010.000000   313.000000

[8 rows x 6 columns]